File: week1/18258.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        Deque<Integer> q = new ArrayDeque<>();
        int n = Integer.parseInt(br.readLine());
        StringBuilder sb = new StringBuilder();

        while (n-- > 0) {
            String[] cmd = br.readLine().split(" ");
            switch (cmd[0]) {
                case "push": q.offer(Integer.parseInt(cmd[1])); break;
                case "pop": sb.append(q.isEmpty() ? -1 : q.poll()).append("\n"); break;
                case "size": sb.append(q.size()).append("\n"); break;
                case "empty": sb.append(q.isEmpty() ? 1 : 0).append("\n"); break;
                case "front": sb.append(q.isEmpty() ? -1 : q.peek()).append("\n"); break;
                case "back": sb.append(q.isEmpty() ? -1 : ((ArrayDeque<Integer>) q).peekLast()).append("\n"); break;
            }
        }
        System.out.print(sb);
    }
}

File: week1/SUNWOO/18258.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    int N;
    cin >> N;
    queue<int> q;

    while (N--) {
        string cmd;
        cin >> cmd;

        if (cmd == "push") {
            int x;
            cin >> x;
            q.push(x);
        }
        else if (cmd == "pop") {
            if (q.empty()) cout << -1 << '\n';
            else {
                cout << q.front() << '\n';
                q.pop();
            }
        }
        else if (cmd == "size") {
            cout << q.size() << '\n';
        }
        else if (cmd == "empty") {
            cout << q.empty() << '\n';
        }
        else if (cmd == "front") {
            cout << (q.empty() ? -1 : q.front()) << '\n';
        }
        else if (cmd == "back") {
            cout << (q.empty() ? -1 : q.back()) << '\n';
        }
    }

    return 0;
}
